src/App.tsx

```diff
@@ -1,14 +1,52 @@
 /* eslint-disable max-len */
 import React from 'react';
+import { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+  const [selectFilter, setSelectFilter] = useState('all');
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getTodos()
+      .then(data => {
+        setTodos(data);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    const byTitle = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    const statusSelect = (() => {
+      switch (selectFilter) {
+        case 'active':
+          return !todo.completed;
+
+        case 'completed':
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    })();
+
+    return byTitle && statusSelect;
+  });
```

Comment:
The filtering logic is correct, but it could be simplified for better readability. Consider separating the filter conditions into named functions or variables to make the code more self-explanatory.